merkledag/merkledag.go

```diff
@@ -146,6 +146,15 @@ func (sg *sesGetter) GetMany(ctx context.Context, keys []*cid.Cid) <-chan *ipld.
 	return getNodesFromBG(ctx, sg.bs, keys)
 }
 
+func NewSession(ctx context.Context, dag ipld.DAGService) ipld.NodeGetter {
```

Comment:
* This should probably take a NodeGetter.
* Docs are nice. What happens when the context is canceled? What happens to in-flight requests (esp if we just return the dag and not a session)?

Reply:
Alternatively, we could define a new interface:

```go
type Session interface {
	Session(ctx context.Context) ipld.NodeGetter
}
func NewSession(ctx context.Context, dag ipld.NodeGetter) ipld.NodeGetter {
	if ses, ok := dag.(Session); ok {
    	return ses.Session(ctx)
    }
    return dag
}
```